mitodl/mit-open
# Clean up / consolidate channels code (for fields)

# What are the relevant tickets?
#215 

# Description (What does it do?)
This PR:
1. Renames `channels_fields` app to `channels`. This app includes our FieldChannel model, and is a home for future channel variants.
2. Moves BaseChannel to the `channels` app from `discussions`.
3. Removes the old `discussions` app. The old discussions app was very bare-bones after we removed reddit. #48 

# How can this be tested?
There should be not noticeable change in behavior: Tests should pass, and you should still be able to create/edit/view channels.

Assuming you have some learningpaths populated:

1. **Create Fields:** Create some fields at http://localhost:8063/api/v1/fields/
   - You can see some associated docs at http://localhost:8063/api/v1/schema/redoc/#tag/fields/operation/fields_create
   - Check that using non-learning paths for `featured_list` and `lists` gives reasonable errors.
2. **View the field details and Update:** http://localhost:8063/api/v1/fields/<field_name>. In particular, try:
    - assigning re-ordering the lists within a field
    - assigning and re-ordering subfields within a field 

## Notes
You will lose old FieldChannel models from the channels_fields app. I think that's OK for the current stage of this project.
